src/query/functions/src/aggregates/aggregate_array_moving.rs

```diff
@@ -0,0 +1,788 @@
+        let array_value = ScalarRef::Array(inner_col);
+        builder.push(array_value);
+
+        Ok(())
+    }
+}
+
+#[derive(Clone)]
+pub struct AggregateArrayMovingAvgFunction<State> {
+    display_name: String,
+    window_size: Option<usize>,
+    sum_t: PhantomData<State>,
+    return_type: DataType,
+    scale_add: u8,
+}
+
+impl<State> AggregateFunction for AggregateArrayMovingAvgFunction<State>
+where State: SumState
+{
+    fn name(&self) -> &str {
+        "AggregateArrayMovingAvgFunction"
+    }
+
+    fn return_type(&self) -> Result<DataType> {
+        Ok(self.return_type.clone())
+    }
+
+    fn init_state(&self, place: StateAddr) {
+        place.write(|| State::default());
+    }
+
+    fn state_layout(&self) -> Layout {
+        Layout::new::<State>()
+    }
+
+    fn accumulate(
+        &self,
+        place: StateAddr,
+        columns: &[Column],
+        validity: Option<&Bitmap>,
+        _input_rows: usize,
+    ) -> Result<()> {
+        let state = place.get::<State>();
+        state.accumulate(&columns[0], validity)
+    }
+
+    fn accumulate_keys(
+        &self,
+        places: &[StateAddr],
+        offset: usize,
+        columns: &[Column],
+        _input_rows: usize,
+    ) -> Result<()> {
+        State::accumulate_keys(places, offset, &columns[0])
+    }
+
+    fn accumulate_row(&self, place: StateAddr, columns: &[Column], row: usize) -> Result<()> {
+        let state = place.get::<State>();
+        state.accumulate_row(&columns[0], row)
+    }
+
+    fn serialize(&self, place: StateAddr, writer: &mut Vec<u8>) -> Result<()> {
+        let state = place.get::<State>();
+        state.serialize(writer)
+    }
+
+    fn deserialize(&self, place: StateAddr, reader: &mut &[u8]) -> Result<()> {
+        let state = place.get::<State>();
+        state.deserialize(reader)
+    }
+
+    fn merge(&self, place: StateAddr, rhs: StateAddr) -> Result<()> {
+        let rhs = rhs.get::<State>();
+        let state = place.get::<State>();
+        state.merge(rhs)
+    }
+
+    fn merge_result(&self, place: StateAddr, builder: &mut ColumnBuilder) -> Result<()> {
+        let state = place.get::<State>();
+        state.merge_avg_result(builder, 0_u64, self.scale_add, &self.window_size)
+    }
+
+    fn need_manual_drop_state(&self) -> bool {
+        true
+    }
+
+    unsafe fn drop_state(&self, place: StateAddr) {
+        let state = place.get::<State>();
+        std::ptr::drop_in_place(state);
+    }
+}
+
+impl<State> fmt::Display for AggregateArrayMovingAvgFunction<State> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.display_name)
+    }
+}
+
+impl<State> AggregateArrayMovingAvgFunction<State>
+where State: SumState
+{
+    pub fn try_create(
+        display_name: &str,
+        params: Vec<Scalar>,
+        return_type: DataType,
+        scale_add: u8,
+    ) -> Result<AggregateFunctionRef> {
+        let window_size = if params.len() == 1 {
```

Comment:
I think `window_size` using `u64` is enough, if `window_size` is not specified, the default value is `1_u64`.